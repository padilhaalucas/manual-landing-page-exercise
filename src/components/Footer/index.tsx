import React from 'react';
import Image from 'next/image';

const Footer: React.FC = () => {
  return (
    <footer className="bg-sage-green-manual w-full pt-14 px-16 lg:px-4">
      <div className="max-w-6xl mx-auto">
        <div className="flex flex-col md:flex-row justify-between items-center gap-6 md:items-start mb-12">
          <div className="mb-8 md:mb-0">
            <Image src="/assets/logo.svg" alt="Manual logo" width={40} height={40} />
          </div>
          <div className="grid grid-cols-2 md:grid-cols-4 gap-20 md:gap-16">
            <div className="text-dark-green-manual">
              <h5 className="font-semibold tracking-widest text-sm mb-4">PRODUCT</h5>
              <ul className="space-y-2 text-sm">
                {['Popular', 'Trending', 'Guided', 'Products'].map(item => <li key={item} className="py-2">{item}</li>)}
              </ul>
            </div>
            <div className="text-dark-green-manual">
              <h5 className="font-semibold tracking-widest text-sm mb-4">COMPANY</h5>
              <ul className="space-y-2 text-sm">
                {['Press', 'Mission', 'Strategy', 'About'].map(item => <li key={item} className="py-2">{item}</li>)}
              </ul>
            </div>
            <div className="text-dark-green-manual">
              <h5 className="font-semibold tracking-widest text-sm mb-4">INFO</h5>
              <ul className="space-y-2 text-sm">
                {['Support', 'Customer Service', 'Get started'].map(item => <li key={item} className="py-2">{item}</li>)}
              </ul>
            </div>
            <div>
              <h5 className="font-semibold text-dark-green-manual tracking-widest text-sm mb-4">FOLLOW US</h5>
              <div className="flex space-x-4 py-2">
                <a href="#" aria-label="Facebook">
                  <Image src="/assets/facebook.svg" alt="Facebook" width={24} height={24} />
                </a>
                <a href="#" aria-label="Google">
                  <Image src="/assets/google.svg" alt="Google" width={24} height={24} />
                </a>
                <a href="#" aria-label="Twitter">
                  <Image src="/assets/twitter.svg" alt="Twitter" width={24} height={24} />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div className="border-t border-dark-green/20 text-center items-center h-16 flex justify-center text-sm text-dark-green-manual/80">
          <span>
            Â© 2021 Manual. All rights reserved
          </span>
        </div>
      </div>
    </footer>
  );
};

export default Footer;