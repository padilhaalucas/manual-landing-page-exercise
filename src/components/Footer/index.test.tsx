import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Footer from './index';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    return <img {...props} />
  },
}));

describe('Components/Footer', () => {
  beforeEach(() => {
    render(<Footer />);
  });

  it('should render the logo', () => {
    const logo = screen.getByAltText('Manual logo');
    expect(logo).toBeInTheDocument();
  });

  it('should render all section headings', () => {
    const headings = ['PRODUCT', 'COMPANY', 'INFO', 'FOLLOW US'];
    headings.forEach(heading => {
      expect(screen.getByText(heading)).toBeInTheDocument();
    });
  });

  it('should render all product items', () => {
    const items = ['Popular', 'Trending', 'Guided', 'Products'];
    items.forEach(item => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });

  it('should render all company items', () => {
    const items = ['Press', 'Mission', 'Strategy', 'About'];
    items.forEach(item => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });

  it('should render all info items', () => {
    const items = ['Support', 'Customer Service', 'Get started'];
    items.forEach(item => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });

  it('should render social media icons', () => {
    const socialMedia = ['Facebook', 'Google', 'Twitter'];
    socialMedia.forEach(platform => {
      const icon = screen.getByAltText(platform);
      expect(icon).toBeInTheDocument();
    });
  });

  it('should render the copyright text', () => {
    const copyright = screen.getByText('Â© 2021 Manual. All rights reserved');
    expect(copyright).toBeInTheDocument();
  });

  it('should render the correct number of list items', () => {
    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(11); //
  });

  it('should render the correct number of social media links', () => {
    const socialLinks = screen.getAllByRole('link');
    expect(socialLinks).toHaveLength(3);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});