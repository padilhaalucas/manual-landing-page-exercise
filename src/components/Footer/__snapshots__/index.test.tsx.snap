// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Footer should match snapshot 1`] = `
<footer
  className="bg-sage-green-manual w-full pt-14 px-16 lg:px-4"
>
  <div
    className="max-w-6xl mx-auto"
  >
    <div
      className="flex flex-col md:flex-row justify-between items-center gap-6 md:items-start mb-12"
    >
      <div
        className="mb-8 md:mb-0"
      >
        <img
          alt="Manual logo"
          height={40}
          src="/assets/logo.svg"
          width={40}
        />
      </div>
      <div
        className="grid grid-cols-2 md:grid-cols-4 gap-20 md:gap-16"
      >
        <div
          className="text-dark-green-manual"
        >
          <h5
            className="font-semibold tracking-widest text-sm mb-4"
          >
            PRODUCT
          </h5>
          <ul
            className="space-y-2 text-sm"
          >
            <li
              className="py-2"
            >
              Popular
            </li>
            <li
              className="py-2"
            >
              Trending
            </li>
            <li
              className="py-2"
            >
              Guided
            </li>
            <li
              className="py-2"
            >
              Products
            </li>
          </ul>
        </div>
        <div
          className="text-dark-green-manual"
        >
          <h5
            className="font-semibold tracking-widest text-sm mb-4"
          >
            COMPANY
          </h5>
          <ul
            className="space-y-2 text-sm"
          >
            <li
              className="py-2"
            >
              Press
            </li>
            <li
              className="py-2"
            >
              Mission
            </li>
            <li
              className="py-2"
            >
              Strategy
            </li>
            <li
              className="py-2"
            >
              About
            </li>
          </ul>
        </div>
        <div
          className="text-dark-green-manual"
        >
          <h5
            className="font-semibold tracking-widest text-sm mb-4"
          >
            INFO
          </h5>
          <ul
            className="space-y-2 text-sm"
          >
            <li
              className="py-2"
            >
              Support
            </li>
            <li
              className="py-2"
            >
              Customer Service
            </li>
            <li
              className="py-2"
            >
              Get started
            </li>
          </ul>
        </div>
        <div>
          <h5
            className="font-semibold text-dark-green-manual tracking-widest text-sm mb-4"
          >
            FOLLOW US
          </h5>
          <div
            className="flex space-x-4 py-2"
          >
            <a
              aria-label="Facebook"
              href="#"
            >
              <img
                alt="Facebook"
                height={24}
                src="/assets/facebook.svg"
                width={24}
              />
            </a>
            <a
              aria-label="Google"
              href="#"
            >
              <img
                alt="Google"
                height={24}
                src="/assets/google.svg"
                width={24}
              />
            </a>
            <a
              aria-label="Twitter"
              href="#"
            >
              <img
                alt="Twitter"
                height={24}
                src="/assets/twitter.svg"
                width={24}
              />
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="border-t border-dark-green/20 text-center items-center h-16 flex justify-center text-sm text-dark-green-manual/80"
    >
      <span>
        Â© 2021 Manual. All rights reserved
      </span>
    </div>
  </div>
</footer>
`;
